packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -126,8 +129,15 @@ class RenderCustomMultiChildLayoutBox extends RenderBox
     assert(newDelegate != null);
     if (_delegate == newDelegate)
       return;
+    if (newDelegate == null) {
+      assert(_delegate != null);
+      markNeedsLayout();
+    } else if (_delegate == null ||
```

Comment:
it can't be null, the constructor asserts that it isn't and all subsequent setters are guarded with an assert that it's not null.